mitiq/zne/inference.py

```diff
@@ -920,6 +922,136 @@ def reduce(self) -> float:
+
+    Args:
+        scale_factors: Sequence of noise scale factors at which
+                       expectation values should be measured.
+        shot_list: Optional sequence of integers corresponding to the number
+                   of samples taken for each expectation value. If this
+                   argument is explicitly passed to the factory, it must have
+                   the same length of scale_factors and the executor function
+                   must accept "shots" as a valid keyword argument.
+
+    Raises:
+        ValueError: If data is not consistent with the extrapolation model.
+        ExtrapolationWarning: If the extrapolation fit is ill-conditioned.
+    """
+
+    @staticmethod
+    def extrapolate(
+        scale_factors: Sequence[float],
+        exp_values: Sequence[float],
+        full_output: bool = False,
+    ) -> Union[
+        float,
+        Tuple[
+            float,
+            Union[float, None],
+            List[float],
+            Union[np.ndarray, None],
+            Callable[[float], float],
+        ],
+    ]:
+        # Richardson extrapolation is a particular case of a polynomial fit
+        # with order equal to the number of data points minus 1.
+        # In this version of Richardson extrapolation we map the samples to
+        # a new set of fake nodes and polynomially extrapolate.
+
+        order = len(scale_factors) - 1
+
+        # Define interval [a, b] for which the scale_factors are mapped to
+        a = 1.0
+        b = scale_factors[-1]
+
+        # Mapping to the fake nodes
+        _S_scale_factors = RungeFactory._map(scale_factors, a, b)
+
+        opt_params, params_cov = mitiq_polyfit(
+            _S_scale_factors, exp_values, order
+        )
+
+        poly = np.poly1d(opt_params)
+        zne_limit = poly(RungeFactory._map(0.0, a, b))
+
+        if not full_output:
+            return zne_limit
+
+        zne_error = None
+
+        if params_cov is not None:
+            if params_cov.shape == (order + 1, order + 1):
+                zne_error = np.sqrt(params_cov[order, order])
+
+        def zne_curve(scale_factor: float) -> float:
+            return np.polyval(opt_params, scale_factor)
+
+        return zne_limit, zne_error, opt_params, params_cov, zne_curve
+
+    def reduce(self) -> float:
+        """Evaluates the zero-noise limit found by applying the fake node
+        Richardson extrapolation to the internal data stored in the factory.
+
+        Returns:
+            The zero-noise limit.
+        """
+
+        (
+            self._zne_limit,
+            self._zne_error,
+            self._opt_params,
+            self._params_cov,
+            self._zne_curve,
+        ) = self.extrapolate(  # type: ignore
+            self.get_scale_factors(),
+            self.get_expectation_values(),
+            full_output=True,
+        )
+
+        self._already_reduced = True
+        return self._zne_limit
+
+    @staticmethod
+    def _map(
```

Comment:
```suggestion
    def _map_to_fake_nodes(
```
A more explicit name. Note: if renamed, also successive calls to this function should be updated.